src/asm2wasm.h

```diff
@@ -735,11 +735,12 @@ class Asm2WasmBuilder {
   }
 
   // Some conversions might trap, so emit them safely if necessary
-  Expression* makeTrappingFloatToInt(Expression* value) {
+  Expression* makeTrappingFloatToInt(bool signed_, Expression* value) {
     if (trapMode == TrapMode::Allow) {
       auto ret = allocator.alloc<Unary>();
       ret->value = value;
-      ret->op = ret->value->type == f64 ? TruncSFloat64ToInt32 : TruncSFloat32ToInt32;
+      ret->op = signed_ ? (ret->value->type == f64 ? TruncSFloat64ToInt32 : TruncSFloat32ToInt32)
```

Comment:
Nit: Nested ternary expressions are generally unfriendly to read. Though this one isn't too bad, breaking this into an `if`/`else` might be more quickly scannable.

Nit: Mild repetition of `ret->value->type == f64`, maybe promote to a `bool isDouble`?